### PR TITLE
Remove unnecessary null checks in `flutter/material`

### PR DESCRIPTION
**Pull request by goderbauer**
Part of https://github.com/flutter/flutter/issues/118837.

Dart 3 drops support for non-null safe code, so we can finally turn on the unnecessary_null_comparison lint and remove the unnecessary checks it flags.

First commit contains the automated fixes from `dart fix --apply`. Second commit are some manual fixes on top of that.

### CODE REVIEW

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -2198,25 +2164,6 @@ class _MonthItem extends StatefulWidget {
...
-  /// animation smoother and setting it to [DragStartBehavior.down] will make
-  /// drag behavior feel slightly more reactive.
-  ///
-  /// By default, the drag start behavior is [DragStartBehavior.start].
-  ///
-  /// See also:
-  ///
-  ///  * [DragGestureRecognizer.dragStartBehavior], which gives an example for
-  ///    the different behaviors.
-  final DragStartBehavior dragStartBehavior;
```
Comment (cbracken) [inline packages/flutter/lib/src/material/date_picker.dart:2218]: Interesting - this was totally unused other than initially setting it and checking it was non-null? Nice catch!

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 2]
```diff
@@ -1574,7 +1558,7 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
       final double t = decoration.floatingLabelProgress;
       // The center of the outline border label ends up a little below the
       // center of the top border line.
-      final bool isOutlineBorder = decoration.border != null && decoration.border.isOutline;
+      final bool isOutlineBorder = decoration.border.isOutline;
```
Comment (cbracken) [inline packages/flutter/lib/src/material/input_decorator.dart:1561]: Note to self to go back and check decoration.border is now non-nullable.

**Review by cbracken — APPROVED** [review]
LGTM covering from the beginning to `material/tab_indicator.dart`.

LGTM stamp from a Japanese personal seal

**Review by pdblasi-google — APPROVED** [review]
LGTM. Reviewed `tabs.dart` to `typography.dart`.

File: packages/flutter/lib/src/material/input_decorator.dart (hunk 2 above)
Comment (cbracken) [reply packages/flutter/lib/src/material/input_decorator.dart:1561]: And it is:
https://github.com/flutter/flutter/pull/119022/files/c071acf07b6d8b18000796c60d1b8d3b749d447d#diff-478235b366f9f8e527d852fd87629c538ed2123c6c80c21113a4d943b2ab15cfR599

**Pull request merged**